CMakeLists.txt

```diff
@@ -0,0 +1,329 @@
+    if(NOT OpenCV_FOUND OR NOT OPENCV_VIZ_FOUND)
+	    message("Building OpenCV")
+        include(cmake/GetOpenCV_Viz.cmake)
+        set(USE_V4_CODE Yes)
+        add_dependencies(main${ILLIXR_BUILD_SUFFIX}.exe OpenCV_Viz)
+        set(OpenCV_DEP_STR OpenCV_Viz)
+        set(INTERNAL_OPENCV ${OpenCV_DIR})
+        set(BUILD_OPENCV Yes)
+    endif()
+else()
+    find_package(OpenCV REQUIRED)# CONFIG)
+endif()
+
+if(DEFINED OpenCV_VERSION_MAJOR)
+    if (${OpenCV_VERSION_MAJOR} EQUAL 4)
+        set(USE_V4_CODE Yes)
+    else()
+        set(USE_V4_CODE No)
+    endif()
+endif()
+
+find_package(GLEW REQUIRED)
+
+pkg_check_modules(glu REQUIRED glu)
+
+find_package(SQLite3 REQUIRED)
+
+find_package(X11 REQUIRED)
+
+pkg_check_modules(gl REQUIRED gl)
+
+find_package(Eigen3 REQUIRED)
+
+#############
+# plugin specific dependencies
+#############
+
+find_package(Git REQUIRED)
+
+if (BUILD_GROUP STREQUAL "TESTS")
+    find_package(GTest REQUIRED)
+endif()
+
+find_package(Boost REQUIRED COMPONENTS filesystem serialization)
+
+set(EXTERNAL_LIBRARIES "")
+
+# check for 3rd party dependencies that are not parts of OS repos
+macro(get_external proj)
+    list(APPEND EXTERNAL_LIBRARIES "${proj}")
+    set(${proj}_EXTERNAL No)
+    set(${proj}_DEP_STR "")
+    include(${CMAKE_SOURCE_DIR}/cmake/Get${proj}.cmake)
+endmacro()
+
+macro(get_external_for_plugin proj)
+    list(APPEND EXTERNAL_LIBRARIES "${proj}")
+    set(${proj}_EXTERNAL No PARENT_SCOPE)
+    set(${proj}_DEP_STR "" PARENT_SCOPE)
+    include(${CMAKE_SOURCE_DIR}/cmake/Get${proj}.cmake)
+    set(EXTERNAL_LIBRARIES ${EXTERNAL_LIBRARIES} PARENT_SCOPE)
+    set(${proj}_EXTERNAL ${${proj}_EXTERNAL} PARENT_SCOPE)
+    set(${proj}_DEP_STR ${${proj}_DEP_STR} PARENT_SCOPE)
+endmacro()
+
+
+if(USE_GTSAM_INTEGRATOR OR USE_KIMERA_VIO)
+    get_external(GTSAM)
+endif()
+
+target_include_directories(main${ILLIXR_BUILD_SUFFIX}.exe PUBLIC ${X11_INCLUDE_DIR} ${GLEW_INCLUDE_DIR} ${SQLite3_INCLUDE_DIR} ${GLU_INCLUDE_DIR} ${gl_INCLUDE_DIRS} ${OpenCV_INCLUDE_DIRS} ${yaml-cpp_INCLUDE_DIRS} ${CMAKE_SOURCE_DIR}/include)
+target_link_libraries(main${ILLIXR_BUILD_SUFFIX}.exe ${X11_LIBRARIES} ${GLEW_LIBRARIES} ${SQLite3_LIBRARIES} ${glu_LDFLAGS} ${gl_LIBRARIES} ${OpenCV_LIBRARIES} dl pthread stdc++fs ${yaml-cpp_LIBRARIES})
+
+# set any plugins that are being built
+foreach(ITEM IN LISTS PLUGIN_LIST)
+    string(TOUPPER "USE_${ITEM}" ITEM_UPPER)
+    if(${ITEM_UPPER})
+        add_subdirectory(plugins/${ITEM})
+    endif()
+endforeach()
+
+# external plugins
+if(USE_KIMERA_VIO OR USE_ORB_SLAM)
+    get_external(DBoW2)
+endif()
+
+
+foreach(ITEM IN LISTS EXT_PLUGIN_LIST)
+    string(TOUPPER "USE_${ITEM}" ITEM_UPPER)
+    if(${ITEM_UPPER})
+        include(cmake/Get${ITEM}.cmake)
+    endif()
+endforeach()
+
+file(DOWNLOAD ${DATA_FILE} data.zip SHOW_PROGRESS)
```

Comment:
I am also running into issue here, if I use relative path, I am encountering error "file DOWNLOAD error: cannot create directory '' - Specify file by full path", I changed to ./data.zip and it works fine. My cmake version is 3.16.3.

Reply:
Changing that should be ok. In the long term we should have a standard demo data file that is part of the github repo, rather than relying on downloading from somewhere.